chain-signatures/node/src/indexer_eth.rs

```diff
@@ -879,6 +875,113 @@ async fn process_block(
         return Ok(());
     };
 
+    let block_receipts_clone = block_receipts.clone();
+    let pending_txs: HashMap<SignRespondTxId, SignRespondTx> = {
+        sign_respond_tx_map
+            .read()
+            .await
+            .iter()
+            .map(|(id, tx)| (*id, tx.clone()))
+            .filter(|(_, tx)| tx.status == SignRespondTxStatus::Pending)
+            .collect()
+    };
+    let mut read_respond_requests: Vec<IndexedSignRequest> = Vec::new();
+    for receipt in block_receipts_clone {
+        let Some(pending_tx) = pending_txs.get(&receipt.transaction_hash.into()) else {
+            continue;
+        };
+        let status = receipt.status();
+        tracing::info!(
+            "Tx {} found in block {block_number}: {}",
+            receipt.transaction_hash,
+            if status { "success" } else { "failed" }
+        );
+        let client_clone = client.clone();
+        let mut pending_tx = pending_tx.clone();
+        pending_tx.status = if status {
+            SignRespondTxStatus::Success
+        } else {
+            SignRespondTxStatus::Failed
+        };
+
+        let pending_tx_clone = pending_tx.clone();
+        let completed_tx = crate::read_respond::CompletedTx::new(pending_tx_clone, block_number);
+
+        let read_respond_request: Option<IndexedSignRequest> = completed_tx
+            .create_sign_request_from_completed_tx(&client_clone, 6, total_timeout)
+            .await;
+        if let Some(read_respond_request) = read_respond_request {
+            read_respond_requests.push(read_respond_request);
+            sign_respond_tx_map
+                .write()
+                .await
+                .insert(receipt.transaction_hash.into(), pending_tx.clone());
+        } else {
+            // failed to create sign request from completed tx, remove the tx from the map
+            tracing::warn!(
+                "Failed to create sign request from completed tx, removing tx from map: {:?}",
+                receipt.transaction_hash
+            );
+            sign_respond_tx_map
+                .write()
+                .await
+                .remove(&receipt.transaction_hash.into());
+        }
+    }
+
+    let remaining_pending_txs: HashMap<SignRespondTxId, SignRespondTx> = {
+        sign_respond_tx_map
+            .read()
+            .await
+            .iter()
+            .map(|(id, tx)| (*id, tx.clone()))
+            .filter(|(_, tx)| tx.status == SignRespondTxStatus::Pending)
+            .collect()
+    };
+
+    for (tx_id, mut tx) in remaining_pending_txs {
```

Comment:
what's the difference between processing the first set of pending tx then processing remaining ones?